src/fNull.js

```diff
@@ -0,0 +1,46 @@
+import { isNil, curryN } from 'ramda';
+
+import defaultWhen from './defaultWhen';
+import isFunction from './isFunction';
+import isArray from './isArray';
+
+/**
+ * Returns a function which is called with the given arguments. If any of the given arguments are null or undefined,
+ * the corresponding default value for that argument is used instead to call the original function.
+ *
+ * @func fNull
+ * @memberOf RA
+ * @category Function
+ * @sig function, [defaultArgs] -> function -> function(args or defaultArgs)
+ * @param {Function} function to be executed
+ * @param {Array|undefined} default arguments to call the function with
+ * @return {Function} function to be called
+ * @example
+ *
+ * RA.fNull(); // => fn // => undefined
+ * RA.fNull(fn); // => fn // => fn(args) //no default arguments
+ * RA.fNull(fn, [a]) // => fn // => fn(args) first argument with default param as a
+ */
+
+const fNull = curryN(2, (originalFunction, defaultArgs) => {
```

Comment:
Couple of notes here:

- Function needs to be of arity 2 (satisfied)
- Function needs to return function of the same arity as original function (not satisfied)
- Function must be total (not satisfied)
- No parts of the arguments nor return value must be checked (not satisfied)

Here is my proposal of the function. 

```js
const fnull = curry((fn, defaults) => {
  return curryN(fn.length, (...args) => {
    const argsWithDefaults = mapIndexed(
      (val, idx) => defaultWhen(isNil, defaults[idx], val),
      args
    );

    return apply(fn, argsWithDefaults);
  });
});
```
I've tried to use functions that we already have in R or RA.

Reply:
Aah got it after taking my time to understand this code. 
The artiy of the function is important to enable currying for each argument. 
Also, the function did not allow currying at all, the way I had implemented it. 

Learnt something new today (total functions)

Reply:
Right. This is a good place to study the FP jargon: https://github.com/hemanth/functional-programming-jargon